### PR TITLE
[PRODDEV-479] [PRODDEV-480]  Flag new shared content

### PR DESCRIPTION
**Pull request by froboy**
Do not merge PR, only review it and approve/request changes.
Merge is allowed by QA Engineer only.

**Related Issue/Ticket:**

**PLEASE CHECK BASE BRANCH FOR YOUR PR**
**ONLY urgent and approved fixes for point release should go to master branch**

https://openy.atlassian.net/browse/PRODDEV-479
https://openy.atlassian.net/browse/PRODDEV-480

## Steps to test:

You will need to have a site connected to a shared content server in order to test and have the shared content server all configured.

Check out this video: 

https://user-images.githubusercontent.com/238201/145146512-354e3c73-8feb-47c8-98cc-c45ae9312ccd.mp4

- [ ] Observe "Fetch to my site" button is duplicated on the top of the form.
Fetch_shared_content___Drush_Site-Install
- [ ] Observe top "Fetch to my site" button works exactly the same as the bottom button.

- [ ] Observe once content is added to a site, the checkbox and button are disabled and the button text changes from "Preview" to "Added"
Fetch_shared_content___Drush_Site-Install

- [ ] Observe when you click "Preview" a "Fetch to my site" button that fetches content when clicked.
- [ ] Observe close button that closes the preview with no action.

Fetch_shared_content___Drush_Site-Install

- [ ] Observe content is bolded when new (or when it's your first session)
- [ ] Observe content is unbolded when you Preview it and then close the modal.
- [ ] Observe content is unbolded when you start a new session.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [ ] This change does not contain front-end fixes.
- [ ] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

**Comment by froboy** [comment]
@anpolimus I've completed refactoring based on our call this morning. This is now ready for review again.

**Comment by froboy** [comment]
retest

**Comment by AndreyMaximov** [comment]
retest

File: modules/openy_gc_shared_content/src/Form/SharedContentPreviewForm.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    // Add the core AJAX library.
+    $form['#attached']['library'][] = 'core/drupal.ajax';
```
Comment (anpolimus) [inline modules/openy_gc_shared_content/src/Form/SharedContentPreviewForm.php:84]: there is no need to do that. Drupal addsa drupal.ajax to the each form itself.
```suggestion
    
```

**Review by anpolimus — CHANGES_REQUESTED** [review]

File: modules/openy_gc_shared_content/src/Form/SharedContentPreviewForm.php [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    
+
+    // Group submit handlers in an actions element with a key of "actions" so
```
Comment (froboy) [inline modules/openy_gc_shared_content/src/Form/SharedContentPreviewForm.php:85]:
```suggestion
    // Group submit handlers in an actions element with a key of "actions" so
```

**Comment by AnastasiiaPys** [comment]
Tested. Works as expected.
1.Shared Content is flagged as new by formatting the row with “bold text”
2.After user clicked the Preview button, the row looks unbolded
3.Content updated on the server since user (Admin) last visit is flagged as “new” by formatting the row with *bold text*.
4.After the content was fetched to the site, the button “Preview” changed to the “Added”.

But an error occurred  after I added new content to the server and fetched the content again. 
image

@froboy is this the type of error that is preventing me from merging this build? or can i merge the build?

**Comment by froboy** [comment]
@AnastasiiaPys comparing the filename in the first line of the error to the files changed here I can see that the error is definitely a result of my new code, so we should resolve that and retest. I'll get on it. Thanks!

**Comment by froboy** [comment]
@AnastasiiaPys I believe I've resolved that issue. Please retest when you're able. Thanks!

**Review by anpolimus — APPROVED** [review]

**Pull request merged**